MTM-53267 improve microservice scope documentation (#1509)
SoftwareAG/c8y-docs: content/microservice-sdk/java-bundle/developing-microservice.md

}
```

> By default autowiring will use tenant scoped beans:

Reviewer: ```suggestion
By default, auto-wiring uses tenant-scoped beans:
```

```java
@Autowired
private EventApi eventApi;
```

This will inject bean that is in `@TenantScope` and will use service user credentials.

> We can inject `@UserScope` version of this Api using `@Qualifier("userEventApi")` each Api bean has name composing of `user` and name of the api in this case `EventApi` in code it will be:

Reviewer: ```suggestion
We can inject the `@UserScope` version of this API using `@Qualifier("userEventApi")`. The name for each API bean is composed of `user` and the name of the API, in this case `EventApi`. In the code it will look like the following:
```

```java
@Autowired
@Qualifier("userEventApi")
private EventApi eventApi;
```

> Scope of the bean can be overridden by using ContextService for user `ContextService<UserCredentials>` and for tenant `ContextService<MicroserviceCredentials`. By using those ContextServices we can use service user credentials like in `@TenantScope` bean with user credentials or use service user credentials in `@UserScope`, for example:

Reviewer: ```suggestion
The scope of the bean can be overridden by using `ContextService<UserCredentials>` for the user and `ContextService<MicroserviceCredentials` for the tenant. By using these ContextServices we can use the service user credentials like in the `@TenantScope` bean with user credentials or use the service user credentials in `@UserScope`, for example:
```

```java
@Autowired
private PlatformProperties platformProperties;
@Autowired
private ContextService<UserCredentials> contextService;
@Autowired
private EventApi eventApi;

public PagedEventCollectionRepresentation get10Events () {
    return contextService.runWithinContext(contextService.getContext(), () -> eventApi.getEvents().get(10));
}
```

> This function will get events using current user credentials.

Reviewer 2: ```suggestion
This function retrieves events using the current user credentials.
```

> In the next example we have two methods. First one will use user credentials to call `EventApi`. Second uses service user credentials.

Reviewer: ```suggestion
The next example shows two methods. The first one uses user credentials to call `EventApi`. The second one uses service user credentials.
```
